src/Umbraco.Infrastructure/Persistence/Querying/Query.cs

```diff
@@ -49,6 +49,22 @@ public virtual IQuery<T> WhereIn(Expression<Func<T, object>>? fieldSelector, IEn
         return this;
     }
 
+    /// <summary>
+    ///     Adds a where-not-in clause to the query.
+    /// </summary>
+    public virtual IQuery<T> WhereNotIn(Expression<Func<T, object>>? fieldSelector, IEnumerable? values)
+    {
+        if (fieldSelector == null)
```

Comment:
```suggestion
        if (fieldSelector is null)
```

Reply:
Have applied here and the other instances in the `Query.cs` file.